src/components/RegistrationForm.tsx

```diff
@@ -21,81 +26,260 @@ const labelStyle = {
+    const inputStyle = {
+        width: fieldLength,
+        height: fieldHeight,
+        backgroundColor: 'white',
+    };
 
-    const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
-        event.preventDefault();
-        // Registration logic
-        //For testing purpose:
-        navigate('/parent-dashboard');
+    const buttonLength = useBreakpointValue({ base: '120px', md: '150px' });
+    const buttonHeight = useBreakpointValue({ base: '40px', md: '50px' });
+    const buttonStyle = {
+        width: buttonLength,
+        height: buttonHeight,
+        fontWeight: 'bold',
     };
-    const handleCancel = () => {
-        navigate('/');
+
+    const initialValues: RegistrationFormData = {
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: '',
+        confirmPassword: '',
+        role: '',
     };
 
     return (
-        <form onSubmit={handleSubmit}>
-            <Stack spacing={4}>
-                <FormControl isRequired>
-                    <FormLabel htmlFor="firstName" style={labelStyle}>
-                        First Name
-                    </FormLabel>
-                    <Input type="text" id="firstName" name="firstName" style={inputStyle} />
-                </FormControl>
-                <FormControl isRequired>
-                    <FormLabel htmlFor="lastName" style={labelStyle}>
-                        Last Name
-                    </FormLabel>
-                    <Input type="text" id="lastName" name="lastName" style={inputStyle} />
-                </FormControl>
-                <FormControl isRequired>
-                    <FormLabel htmlFor="password" style={labelStyle}>
-                        Password
-                    </FormLabel>
-                    <Input id="password" name="password" type="password" style={inputStyle} />
-                </FormControl>
-                <FormControl isRequired>
-                    <FormLabel htmlFor="confirmPassword" style={labelStyle}>
-                        Confirm Password
-                    </FormLabel>
-                    <Input
-                        id="confirmPassword"
-                        name="confirmPassword"
-                        type="password"
-                        style={inputStyle}
-                    />
-                </FormControl>
-                <RadioGroup id="role" name="role" width="22em">
-                    <Stack direction="row" display="flex" alignItems="center">
-                        <Radio value="parent" flex="1" bg="white" size="md">
-                            <span style={radioLabelStyle}>Parent</span>
-                        </Radio>
-                        <Radio value="tutor" flex="1" bg="white" size="md">
-                            <span style={radioLabelStyle}>Tutor</span>
-                        </Radio>
+        <Formik
+            initialValues={initialValues}
+            validationSchema={Yup.object({
```

Comment:
In file structure there is folder "validationSchemas" move this validation object to index.ts file.

Reply:
Donw